go.mod

```diff
@@ -1,14 +1,14 @@
 module github.com/prometheus/prometheus
 
-go 1.21
+go 1.22.3
```

Comment:
question (non-blocking): Is this change related?

I have nothing against bumping go for the module. But if it's not necessary, doing this in a separate PR or commit will make it easier to isolate these changes as they roll out.

Reply:
`go mod tidy` updates the Go version to match the smaller of the minimum Go version of dependencies OR the current version in `go.mod`.

In this case, it updated to because prometheus-engine was using this version: https://github.com/GoogleCloudPlatform/prometheus-engine/blob/main/go.mod